iot/mqttsnippets/subscribe_gateway_device_topic_test.go

```diff
@@ -44,10 +44,10 @@ func TestSubscribeGatewayToDeviceTopic(t *testing.T) {
 	projectID := testutil.SystemTest(t).ProjectID
 
 	registryID := "golang-iot-test-registry"
-	gatewayID := createIDForTest("gateway")
-	deviceID := createIDForTest("device")
+	gatewayID := createID("gateway")
+	deviceID := createID("device")
 
-	testutil.Retry(t, 1, 10*time.Second, func(r *testutil.R) {
+	testutil.Retry(t, 5, 5*time.Second, func(r *testutil.R) {
```

Comment:
Sorry, I think I was unclear. Should all of these `create` calls be in separate `testutil.Retry` blocks? Try to create the gateway, then try to create device, then try to bind them. And so on. That way, if everything succeeds except binding the device, the test doesn't have to create the gateway/device again. Or maybe `createGateway` and `createDevice` do nothing if they already exist?

Reply:
So as it stands right now, you're right, trying to create a device that already exists will throw an error. I could rewrite the existing code to account for this, but I think wrapping each call in separate `testutil.Retry` probably makes more sense.